From 43ab37cec5bfa53f764beb3247db57055a15e155 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:39:24 -0400
Subject: [ethernet] i40e: use WoL flag when setting LAA

Message-id: <1409658042-22689-204-git-send-email-sassmann@redhat.com>
Patchwork-id: 90395
O-Subject: [RHEL7.1 PATCH 203/281] i40e: use WoL flag when setting LAA
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Shannon Nelson <shannon.nelson@intel.com>

Make sure the Firmware sets up the LAA as a Wake-On-LAN address.

Change-ID: I57b9acd8c288424fcfed0911053eb725c400b41c
Signed-off-by: Shannon Nelson <shannon.nelson@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit cc41222c55e9ec44d148dee67e6ad407e3b79e46)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index 093ac68..ae9622a 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -1337,7 +1337,7 @@ static int i40e_set_mac(struct net_device *netdev, void *p)
  if (vsi->type == I40E_VSI_MAIN) {
   i40e_status ret;
   ret = i40e_aq_mac_address_write(&vsi->back->hw,
-      I40E_AQC_WRITE_TYPE_LAA_ONLY,
+      I40E_AQC_WRITE_TYPE_LAA_WOL,
       addr->sa_data, NULL);
   if (ret) {
    netdev_info(netdev,
-- 
1.7.1